backend/lib/db.js

```diff
@@ -0,0 +1,146 @@
+import assert from 'assert'
+import { ObjectID, MongoClient } from 'mongodb'
+
+import { MongoDBData } from '../../config/projectInfoData.js'
+import { posts_db_name } from '../Utilities/API_utilities'
+class DataBase {
+    constructor () {
+        this.db = {}
+        this.url
+        this.fieldsWanted = {
+            _id: 1,
+            date: 1,
+            type: 1,
+        }
+        this.posts_db_name = posts_db_name
+        if (process.env.NODE_ENV === 'production') {
+            this.url = MongoDBData['mongoData']['productionURL']
+        } else {
+            this.url = MongoDBData['mongoData']['productionURL']
+        }
+        MongoClient.connect(this.url, (err, dbParam) => {
+            assert.equal(null, err)
+            console.log('Successfully connected to MongoDB server. ')
+            this.db = dbParam.db('DisasterResponse')
+            
+            return;
+        })
+
+    }
+
+    getAll(collection, sortParam = {}) {
+        return new Promise((resolve, reject) => {
```

Comment:
Same thing as above - I don't think you need to re-wrap everything in a Promise again.  

You should be able to just do:

```js
return this.db.collection(collection).find({}).sort(sortParam).toArray()
```

`toArray` returns a Promise if no callback is passed in.